src/haz3lcore/dynamics/transition/Transition.re

```diff
@@ -271,61 +271,49 @@ module Transition = (EV: EV_MODE) => {
         kind: FixClosure,
         is_value: false,
       });
+    | FixF({term: Asc(p, t), _}, d1, env) =>
+      // TODO Decide if this is the right approach
+      let. _ =
+        otherwise(env |> Option.value(~default=ClosureEnvironment.empty), d);
+      Step({
+        expr: FixF(p, Asc(d1, t) |> DHExp.fresh, env) |> rewrap,
+        state_update,
+        kind: FixUnwrap,
+        is_value: false,
+      });
```

Comment:
I don't think this is the right approach alone. It's contingent on how we do statics for fixpoints.

If we get the type for a fixpoint from the pattern then `fix a: Int : String -> ?` -> `fix a : Int -> ? : String`. Which means we have something that was a string is now an int violating preservation. So we either want to use the expression inside the  fixpoint to get the type or we need some form of eager pattern matching that results in something of a consistent type. Since ascriptions come off of patterns in reverse the type that it fundamentally has is the most centered one.

Reply:
I'm leaning towards not modifying evaluation much and changing statics to mark the fixpoint as unknown with inconsistent pattern/expression like:
![Screenshot From 2025-05-28 15-59-08](https://github.com/user-attachments/assets/80633ed3-a819-461b-be8d-6a3a42a60fe2)